commercetools-convenience/src/test/java/io/sphere/sdk/sequencegenerators/BigIntegerNumberGeneratorIntegrationTest.java

```diff
@@ -92,6 +100,44 @@ public void concurrentUsageTest() throws Exception {
         assertThat(numbersGenerated.containsAll(expectedNumbers)).isTrue();
     }
 
+    @Test
+    public void failWithNotConcurrentModificationException() {
+        final String container = randomKey();
+        final String key = randomKey();
+        final CustomObjectBigIntegerNumberGeneratorConfig config =
+                CustomObjectBigIntegerNumberGeneratorConfigBuilder.of(getSphereClient(getBadRequestHttpClient()), key)
+                        .container(container)
+                        .build();
+        final BigIntegerNumberGenerator generator = CustomObjectBigIntegerNumberGenerator.of(config);
+        try {
+            generator.getNextNumber().toCompletableFuture().join();
```

Comment:
If this does not throw an exception then the test would be still green.

Have a look at http://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#exception-assertion and use `catchThrowable`.